diff --git a/slime/ray/rollout.py b/slime/ray/rollout.py
index 3c5e918..bafbe52 100644
--- a/slime/ray/rollout.py
+++ b/slime/ray/rollout.py
@@ -136,6 +136,9 @@ class RolloutManager:
     def check_weights(self, action: str):
         return ray.get([engine.check_weights.remote(action=action) for engine in self.rollout_engines])
 
+    def get_weight_versions(self):
+        return ray.get([engine.get_weight_version.remote() for engine in self.rollout_engines])
+
     def _get_rollout_data(self, rollout_id):
         if self.args.load_debug_rollout_data:
             data = torch.load(
diff --git a/slime/backends/sglang_utils/sglang_engine.py b/slime/backends/sglang_utils/sglang_engine.py
index e665f28..9699330 100644
--- a/slime/backends/sglang_utils/sglang_engine.py
+++ b/slime/backends/sglang_utils/sglang_engine.py
@@ -273,7 +273,7 @@ class SGLangEngine:
     def get_weight_version(self):
         if self.node_rank != 0:
             return
-        url = f"http://{self.server_host}:{self.server_port}/get_weight_version"
+        url = f"http://{self.server_host}:{self.server_port}/weight_version"
         response = requests.get(url)
         response.raise_for_status()
         return response.json()["weight_version"]
